corehq/apps/data_dictionary/views.py

```diff
@@ -97,17 +115,35 @@ def data_dictionary_json(request, domain, case_type_name=None):
 @toggles.DATA_DICTIONARY.required_decorator()
 def update_case_property(request, domain):
     property_list = json.loads(request.POST.get('properties'))
+    fhir_resource_type = request.POST.get('fhir_resource_type')
+    fhir_resource_type_obj = None
+    case_type = request.POST.get('case_type')
     errors = []
+    if fhir_resource_type and case_type:
+        case_type_obj = CaseType.objects.get(domain=domain, name=case_type)
+        try:
+            fhir_resource_type_obj = FHIRResourceType.objects.get(case_type=case_type_obj, domain=domain)
+        except FHIRResourceType.DoesNotExist:
+            fhir_resource_type_obj = FHIRResourceType(case_type=case_type_obj, domain=domain)
```

Comment:
Django has a shortcut for this.
```suggestion
        fhir_resource_type_obj = FHIRResourceType.objects.get_or_create(
            domain=domain,
            case_type=case_type_obj,
        )
```

Reply:
Ya, i was trying to avoid two SQL saves. Since I need to update the object's name and then save it anyway. If i was just creating the object, I would have used get_or_create.
Does that sound reasonable?

Reply:
That does make sense.

(I made a mistake in my suggested change: `get_or_create()` returns a tuple, `obj, created`. You could use `created` to figure out if you need to update the existing instance, but I think what you've got here is neater.)